app/models/status.rb

```diff
@@ -103,7 +103,11 @@ class Status < ApplicationRecord
   scope :not_domain_blocked_by_account, ->(account) { account.excluded_from_timeline_domains.blank? ? left_outer_joins(:account) : left_outer_joins(:account).where('accounts.domain IS NULL OR accounts.domain NOT IN (?)', account.excluded_from_timeline_domains) }
   scope :tagged_with_all, lambda { |tag_ids|
     Array(tag_ids).map(&:to_i).reduce(self) do |result, id|
-      result.joins("INNER JOIN statuses_tags t#{id} ON t#{id}.status_id = statuses.id AND t#{id}.tag_id = #{id}")
+      join_query_string = <<~SQL.squish
+        INNER JOIN statuses_tags t:tag_id ON t:tag_id.status_id = statuses.id AND t:tag_id.tag_id = :tag_id
+      SQL
+      sanitized_join_query = sanitize_sql_array([join_query_string, tag_id: id])
+      result.joins(sanitized_join_query)
```

Comment:
The "using it on a `joins` instead of a `where`" part, or the "using it to name our join tables in the sql query" part?

Reply:
Both, but mostly the second part. It's used in the middle of a join table name.

Reply:
Ah, very nice. I think in that case you might not need the join table rename, since the query sits inside the `EXISTS` query and won't collide with the top level query? Will experiment with that locally and push something.

Reply:
Pushed update with the `EXISTS` approach -- my local testing didn't need the join table rename so I removed it ... but let me know if I'm missing something there.

Here's some before/after on the query, from test logs:

From `main` branch, single tag:

```sql
SELECT "statuses"."id" FROM "statuses"
INNER JOIN statuses_tags t6 ON t6.status_id = statuses.id AND t6.tag_id = 6
WHERE "statuses"."deleted_at" IS NULL
ORDER BY "statuses"."id" ASC
```

PR branch, single tag:

```sql
SELECT "statuses"."id" FROM "statuses"
WHERE "statuses"."deleted_at" IS NULL
AND (EXISTS(SELECT 1 FROM statuses_tags WHERE statuses_tags.status_id = statuses.id AND statuses_tags.tag_id = 213))
ORDER BY "statuses"."id" ASC
```

From `main` branch, multiple tags:

```sql
SELECT "statuses"."id" FROM "statuses"
INNER JOIN statuses_tags t32 ON t32.status_id = statuses.id AND t32.tag_id = 32
INNER JOIN statuses_tags t33 ON t33.status_id = statuses.id AND t33.tag_id = 33
WHERE "statuses"."deleted_at" IS NULL
ORDER BY "statuses"."id" ASC
```

PR branch, multiple tags:

```sql
SELECT "statuses"."id" FROM "statuses"
WHERE "statuses"."deleted_at" IS NULL
AND (EXISTS(SELECT 1 FROM statuses_tags WHERE statuses_tags.status_id = statuses.id AND statuses_tags.tag_id = 215))
AND (EXISTS(SELECT 1 FROM statuses_tags WHERE statuses_tags.status_id = statuses.id AND statuses_tags.tag_id = 216))
ORDER BY "statuses"."id" ASC
```